app/src/ui/add-repository/licenses.ts

```diff
@@ -88,6 +88,7 @@ export async function writeLicense(
     'project',
     'description',
     'year',
+    '\n'
```

Comment:
I'm not entirely sure this is the right fix, given how this array of values is used in `replaceTokens`:

 - these license templates expect a `[token]` that they replace with a value from `ILicenseFields`
 - these tokens are constrained to the expected fields from `ILicenseFields`, and the compiler should catch this:

<img width="713" src="https://user-images.githubusercontent.com/359239/54220116-6c747000-44cf-11e9-83e9-ad9d769c0851.png">

 - I think what we need to do is verify the generated license text is ending in `\n`, and if not append a character. This could just be a simple function that checks the string and either returns it or modifies it:

```ts
export function ensureLineEndingSet(body: string) {
  if (body.endsWith('\n')) {
    return body
  } else {
    return `${body}\n`
  }
}
```

 - this could be then used when writing the updated license text which is stored in the `body` variable:

```ts
await writeFile(fullPath, ensureLineEndingSet(body))
```

This also reminded me of how we touch `.gitignore` files, that might end in `\r\n` or `\n` depending on your setup. We have some code lying around to do this correctly, and likely some tests we can use, but for now let's just focus on getting `\n` appended to the generated license.

Reply:
@shiftkey I was quite unaware of the context here :confused:  Thank you for throwing light into the present scenario :+1: